.github/workflows/flowzone.yml

```diff
@@ -490,21 +490,47 @@ jobs:
     name: NPM draft
     runs-on: ubuntu-latest
     needs: [npm_matrix, source_check]
-    if:
-      needs.source_check.outputs.npm_private != 'true' &&
-      inputs.npmjs_repository != ''
+    if: inputs.npmjs_repository != ''
 
     defaults:
       run:
-        working-directory: .
+        working-directory: ${{ inputs.working_directory }}
         shell: bash
 
     steps:
+      - name: Download versioned source
+        uses: actions/download-artifact@v3
+        with:
+          name: versioned-source-${{ github.event.pull_request.head.sha }}
+
+      - name: Extract versioned source
+        working-directory: .
+        run: tar -xvf versioned-source.tar
+
       - name: Publish draft tag
         run: |
-          echo "not yet implemented!"
+          echo '//${{ inputs.npmjs_repository }}/:_authToken=${{ secrets.NPM_TOKEN }}' > ~/.npmrc
+
+          npm whoami
+
+          branch="$(echo '${{ github.event.pull_request.head.ref }}' | sed 's/\//-/g')"
+
+          version="$(cat < ../VERSION)"
 
-      # TODO: publish to npmjs
+          # TBC: we could suffix with ${{ github.event.pull_request.head.sha }},
```

Comment:
@klutchell @Page- what do we want our draft NPM packages to be called?

Reply:
Use the git hash like you suggested. The docker flow uses the git hash, the branch name, and the pr number as tags.

Reply:
Though if version is always required for npm, then maybe with a version prefix, I think that's how resinCI does it?

Reply:
er, {version}-{sha}` as you said

Reply:
how would we clean up all of the SHAs generated by a PR (incl. rebases, etc.), since `${{ github.event.pull_request.head.sha }}` will always be different on a rebase?